v3/client_sortkey.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 U. Cirello (cirello.io and github.com/cirello-io)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamolock
+
+import (
+	"context"
+	"errors"
+)
+
+// ClientWithSortKey is a dynamoDB based distributed lock client, but with a required sort key.
+type ClientWithSortKey struct{ *internalClient }
+
+// NewWithSortKey creates a new dynamoDB based distributed lock client.
+func NewWithSortKey(dynamoDB DynamoDBClient, tableName, sortKeyName string, opts ...ClientOption) (*ClientWithSortKey, error) {
+	if sortKeyName == "" {
+		return nil, errors.New("a sortKeyName must be supplied; use `Client` if you don't want a sort key")
+	}
+
+	internalClient, err := newInternal(dynamoDB, tableName, opts...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &ClientWithSortKey{internalClient}, nil
+}
+
+// AcquireLockWithContext holds the defined lock. The given context is passed
+// down to the underlying dynamoDB call.
+func (c *ClientWithSortKey) AcquireLockWithContext(ctx context.Context, partitionKey, sortKey string, opts ...AcquireLockOption) (*Lock, error) {
```

Comment:
Now that we are jumping to v3, I'd say we could make a breaking change and remove the `WithContext` suffix everywhere. Let `{{Method}}WithContext(context.Context, ...)` just be `{{Method}}(context.Context, ...)`

Reply:
sounds good, i'll do for the next PR